cmd/query/app/server_test.go

```diff
@@ -53,7 +53,20 @@ func TestCreateTLSServerError(t *testing.T) {
 	}
 
 	_, err := NewServer(zap.NewNop(), &querysvc.QueryService{},
-		&QueryOptions{TLS: tlsCfg}, opentracing.NoopTracer{})
+		&QueryOptions{TLSGrpc: tlsCfg}, opentracing.NoopTracer{})
+	assert.NotNil(t, err)
+}
+
+func TestCreateTLSHttpServerError(t *testing.T) {
+	tlsCfg := tlscfg.Options{
+		Enabled:      true,
+		CertPath:     "invalid/path",
+		KeyPath:      "invalid/path",
+		ClientCAPath: "invalid/path",
+	}
+
+	_, err := NewServer(zap.NewNop(), &querysvc.QueryService{},
```

Comment:
Could you also add a positive test, to make sure that the typical path for `queryOptions.TLSHttp.Enabled` is tested?

Reply:
I am sorry,  I am new to this.  Can you show me a similar example (preferably one that involves dummy certificat ,key and other TLSConfig)?   Thanks so much.

Reply:
You can generate a cert locally and place in a test data directory. You can take a look at these examples: 

- ./pkg/config/tlscfg/testdata
- ./cmd/agent/app/reporter/grpc/testdata

Reply:
`httpServer.serveTLS(...)`  that i used from [here](https://golang.org/pkg/net/http/#ServeTLS)  requires  " If the certificate is signed by a certificate authority, the certFile should be the concatenation of the server's certificate, any intermediates, and the CA's certificate. "    Is this a fair assumption?  It is not the case in the testData provided.   The other alternative is to use tls.listener that wraps around the current listener `httpListener`.  This can use TLSConfig directly.

Reply:
I'm pretty sure the certs from the test data are self-signed. For the tests, this is OK and you can use the same CA on both the client and server, so that they both have the same chain of trust.

Reply:
What server name should i use for generating keys?  i have used "localhost" to generate the keys and all requests are made to "localhost".   Can you please tell me the actual server name?  Where can i find the convention ? Thanks.

Reply:
we already have test certificates generated in `./pkg/config/tlscfg/testdata/`

Reply:
And what is the server name for those certificates?  TLS client checks the server name...  Else there could be a man-in-the-middle attack.   `./cmd/agent/app/reporter/grpc/testdata` could be a better set of certificates (client and server are separate certificates)  but that has a fixed server name of `server.jaeger.io`.  p.s.  I am checking TLS and mTLS so need both client and server certifs.

Reply:
I am not sure. We also have another set of test certificates under `./cmd/agent/app/reporter/grpc/testdata`, and they are used for the actual client/server test in `TestProxyClientTLS -- ./cmd/agent/app/reporter/grpc/builder_test.go`.

@tcolgate do you remember how you generated those certificates? We should have documented it somewhere, either in testdata/README or have a Makefile target.

Reply:
About the server name: with gRPC, you can specify what's the expected server name in the cert, like what we do in the flag `--reporter.grpc.tls.server-name`

Reply:
Yes,  but @yurishkuro agreed on Gitter that it would be better to use one set of certificates and one generic servername associated with the certificate (the servername associated with the server's certificate is tested by the client).  I am waiting for someone to tell me what generic name to use so that i can create a client and server set of Certificates and keys with a common CA.

p.s. the expected server name of  `--reporter.grpc.tls.server-name` is actually `server.jaeger.io`  I traced the add flags call from the builder to the actual addition of flags in `tlscfg` and couldnt find a line of thought as to why it was named as `server.jaeger.io` .  I guess some design choise was made that  I am unable to figure out.  If possible,  can you please explain the naming so that i could follow the same convention in the future?  Thanks in advance.

Reply:
When doing an HTTP request, the server will send a TLS cert which includes the subject (ie, who can use the cert). The client then compares the hostname in the HTTP request with the subject in the TLS cert. The flag `server-name` tells gRPC to use the flag's value instead of the value from the HTTP request. This is useful when you make a request to `http://localhost/path` but you know the cert was issued for `server.jaeger.io`.

All in all, `server.jaeger.io` is just a name (an unfortunate one, if you ask me), and it represents a host name including the base domain name (`jaeger.io`, which is registered to an entity foreign to this project). It would probably have been better to name it something like `jaeger-collector.example.com` (see [RFC 2606](https://tools.ietf.org/html/rfc2606) on why `example.com` should be used).